controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,686 @@
+
+	logrus.Infof("Checking existing cluster [%s] status for config [%s]", config.Spec.ClusterName, config.Spec.DisplayName)
+	upstreamSpec, err := buildUpstreamClusterState(ctx, config)
+	if err != nil {
+		return config, err
+	}
+	updateUpstreamClusterState(ctx, config, upstreamSpec)
+
+	logrus.Infof("Cluster [%s] status for config [%s] was checked", config.Spec.ClusterName, config.Spec.DisplayName)
+
+	return config, nil
+}
+
+func (h *Handler) ValidateConfig(config *aksv1.AKSClusterConfig) error {
+	// Check for existing aksclusterconfigs with the same display name
+	aksConfigs, err := h.aksCC.List(config.Namespace, v1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot list aksclusterconfigs for display name check")
+	}
+	for _, c := range aksConfigs.Items {
+		if c.Spec.DisplayName == config.Spec.DisplayName && c.ClusterName != config.ClusterName {
+			return fmt.Errorf("cannot create cluster [%s] because an aksclusterconfig exists with the same name", config.Spec.DisplayName)
+		}
+	}
+
+	// validate nodegroup version
+	if !config.Spec.Imported {
+		cannotBeNilError := "field [%s] cannot be nil for non-import cluster [%s]"
+		if config.Spec.ResourceLocation == "" {
+			return fmt.Errorf(cannotBeNilError, "resourceLocation", config.ClusterName)
+		}
+		if config.Spec.ResourceGroup == "" {
+			return fmt.Errorf(cannotBeNilError, "resourceGroup", config.ClusterName)
+		}
+		if config.Spec.ClusterName == "" {
+			return fmt.Errorf(cannotBeNilError, "clusterName", config.ClusterName)
+		}
+		if config.Spec.BaseURL == "" {
+			return fmt.Errorf(cannotBeNilError, "baseUrl", config.ClusterName)
+		}
+		if config.Spec.AuthBaseURL == "" {
+			return fmt.Errorf(cannotBeNilError, "authBaseUrl", config.ClusterName)
+		}
+		if config.Spec.SubscriptionID == "" {
+			return fmt.Errorf(cannotBeNilError, "subscriptionId", config.ClusterName)
+		}
+		if config.Spec.ClientID == "" {
+			return fmt.Errorf(cannotBeNilError, "clientId", config.ClusterName)
+		}
+		if config.Spec.ClientSecret == "" {
+			return fmt.Errorf(cannotBeNilError, "clientSecret", config.ClusterName)
+		}
+	}
+	if config.Spec.KubernetesVersion != "" {
+		if !isKubernetesVersion(config.Spec.KubernetesVersion) {
+			return fmt.Errorf("Kubernetes version [%s] is not supported for cluster [%s]", config.Spec.KubernetesVersion, config.Spec.ClusterName)
+		}
+	}
+
+	return nil
+}
+
+// buildUpstreamClusterState
+func buildUpstreamClusterState(ctx context.Context, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfigSpec, error) {
+	upstreamSpec := &aksv1.AKSClusterConfigSpec{}
+
+	upstreamSpec.Imported = true
+
+	azureAuthorizer, _ := newClientAuthorizer(config.Spec)
+	resourceClusterClient, _ := newClustersClient(azureAuthorizer, config.Spec)
+	clusterState, err := resourceClusterClient.Get(ctx, config.Spec.ResourceGroup, config.Spec.ClusterName)
+
+	// set kubernetes version
+	upstreamVersion := clusterState.KubernetesVersion
+	if upstreamVersion == nil {
```

Comment:
nit: Do we need a new variable here? I cannot tell if `upstreamVersion` was used again.